lib/commands/intent.js

```diff
@@ -0,0 +1,274 @@
+ * Check https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/WindowConfiguration.java
+ * for more details on possible windowing modes (constants starting with `WINDOWING_MODE_`).
+ * @property {?number|string} activityType - The activity type to launch the activity as.
+ * Check https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/WindowConfiguration.java
+ * for more details on possible activity types (constants starting with `ACTIVITY_TYPE_`).
+ * @property {?number|string} display - The display identifier to launch the activity into.
+ * @property {?string} action - Action name
+ * @property {?string} uri - Unified resource identifier
+ * @property {?string} mimeType - Mime type
+ * @property {?string} identifier - Optional identifier
+ * @property {?string|Array<string>} categories - One or more category names
+ * @property {?string} component - Component name
+ * @property {Array<string|Array<string>>} extras - Optional intent arguments. Must be represented
+ * as array of arrays, where each subarray item contains two items: value type and the value itself.
+ * Supported value types are:
+ * - s: string. Value must be a valid string
+ * - sn: null. Value is ignored for this type
+ * - z: boolean. Value must be either `true` or `false`
+ * - i: integer. Value must be a valid 4-byte integer number
+ * - l: long. Value must be a valid 8-byte long number
+ * - f: float: Value must be a valid float number
+ * - u: uri. Value must be a valid uniform resource identifier string
+ * - cn: component name. Value must be a valid component name string
+ * - ia: Integer[]. Value must be a string of comma-separated integers
+ * - ial: List<Integer>. Value must be a string of comma-separated integers
+ * - la: Long[]. Value must be a string of comma-separated long numbers
+ * - lal: List<Long>. Value must be a string of comma-separated long numbers
+ * - fa: Float[]. Value must be a string of comma-separated float numbers
+ * - fal: List<Float>. Value must be a string of comma-separated float numbers
+ * - sa: String[]. Value must be comma-separated strings. To embed a comma into a string,
+ * escape it using "\,"
+ * - sal: List<String>. Value must be comma-separated strings. To embed a comma into a string,
+ * escape it using "\,"
+ * For example: [['s', 'My String1'], ['s', 'My String2'], ['ia', '1,2,3,4']]
+ */
+
+/**
+ * Starts the given activity intent.
+ *
+ * @param {StartActivityOptions} opts
+ * @returns {string} The command output
+ * @throws {Error} If there was a failure while starting the activity
+ * or required options are missing
+ */
+commands.mobileStartActivity = async function mobileStartActivity (opts = {}) {
+  const {
+    intent,
+    user,
+    wait,
+    stop,
+    windowingMode,
+    activityType,
+    display,
+  } = requireOptions(opts, ['intent']);
+  const cmd = [
+    'am', 'start-activity',
```

Comment:
Should we keep `start` so far?

I should check which version, but an Android  5 emulator needed `start`. Newer ones still support `start`, so `start` is better so far.
(I noticed `startservice` vs `start-service` as well... let me check emulators.)

Reply:
Android 7 was still `start`. 8 was `start-activity`.
We can use `start-activity` when we stop supporting Android 7 and lower, I think.
(Android 11 was able to recognize `start` as start-service)

I also think a comment about this with `-` or not depends on Android version will help out future work. (the below start/stop services as well)

Reply:
Oh, I didn't know it was changed between Android versions. Thanks for the hint. Fixed now.